service/src/main/java/org/cbioportal/service/impl/PatientServiceImpl.java

```diff
@@ -20,13 +22,17 @@
     private PatientRepository patientRepository;
     @Autowired
     private StudyService studyService;
+    @Value("${authenticate:false}")
+    private String AUTHENTICATE;
 
     @Override
     @PostFilter("hasPermission(filterObject, 'read')")
     public List<Patient> getAllPatients(String keyword, String projection, Integer pageSize, Integer pageNumber,
             String sortBy, String direction) {
         
-        return patientRepository.getAllPatients(keyword, projection, pageSize, pageNumber, sortBy, direction);
+        List<Patient> patients = patientRepository.getAllPatients(keyword, projection, pageSize, pageNumber, sortBy, direction);
+        // copy the list before returning so @PostFilter doesn't taint the list stored in the mybatis second-level cache
+        return (AUTHENTICATE.equals("false")) ? patients : new ArrayList<Patient>(patients);
```

Comment:
@ersinciftci @n1zea144 should we worry about the memory footpring of copying the result set?  Is this a very large result set?

Reply:
It shouldn't be very large, not in the scale of millions (for now), and the objects don't have many fields in it.